src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -953,6 +954,11 @@ splitAt n ls
           where
             (xs', xs'') = splitAt' (m - 1) xs
 
+-- GHC parser does not accept {-# ANN type [] NoSpecConstr #-}, so we need
+-- to make a newtype.
+{-# ANN type List NoSpecConstr #-}
+newtype List a = List {getList :: [a]}
```

Comment:
Is newtype wrapping `[a]` enough to stop spec constr ?

Reply:
Its the `NoSpecConstr` annotation that stops it.